### PR TITLE
Improve translation

### PR DESCRIPTION
**Pull request by fitn**
Add translatable and translation traits to avoid copy/paste each time we need to add translation entities.

This PR also adds the possibility for a translated entity to handle translated properties and linked entities

### CODE REVIEW

File: Entity/TranslatableTrait.php [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    protected $locale;
+
+    /**
+     * Gets translation for current locale
+     *
+     * @return TranslatableTrait
+     */
+    public function getTranslation($locale = null)
+    {
+        $locale = ($locale) ? $locale : $this->locale;
```
Comment (jmleroux) [inline Entity/TranslatableTrait.php]: Could be more strict, for example for `$locale = ' '`

File: Updater/Updater.php [hunk 2]
```diff
@@ -44,9 +63,128 @@ public function update($referenceData, array $data, array $options = [])
...
+     *
+     * @param ReferenceDataInterface $referenceData
+     * @param string $propertyPath
+     * @param mixed $value
+     *
+     * @throws \InvalidArgumentException
+     */
+    protected function updateTranslation(ReferenceDataInterface $referenceData, $propertyPath, $value)
+    {
+        $translationPattern = '/^(?<property>[a-zA-Z0-9-_]+)-(?<locale>[a-z]{2}_[A-Z]{2})$/';
```
Comment (jmleroux) [inline Updater/Updater.php]: `[a-zA-Z0-9_-]` is safer to avoid to create an interval

File: Updater/Updater.php [hunk 3]
```diff
@@ -44,9 +63,128 @@ public function update($referenceData, array $data, array $options = [])
...
+     * @param string $propertyPath
+     * @param mixed $value
+     *
+     * @throws EntityNotFoundException
+     */
+    protected function updateAssociatedEntity(ReferenceDataInterface $referenceData, $propertyPath, $value)
+    {
+        $associationMapping = $this->getAssociationMapping($referenceData, 'translations');
+        $associationRepo  = $this->em->getRepository($associationMapping['targetEntity']);
+        $associatedEntity = $associationRepo->findOneBy(['code' => $value]);
```
Comment (jmleroux) [inline Updater/Updater.php]: Will it be correctly cleaned afterward? (to avoid memory leaks)

Comment (fitn) [inline Updater/Updater.php]: Memory leak and cascade detach is an integration concern and should be defined on the reference data doctrine mapping itself.
Btw, if I detach before persisting the entity, it cannot work as the uow will search to persist something it does not know anymore :)

**Pull request merged**